ring/partition_instance_ring.go

```diff
@@ -74,15 +90,24 @@ func (r *PartitionInstanceRing) GetReplicationSetsForOperation(op Operation) ([]
 
 // GetReplicationSetForPartitionAndOperation returns a ReplicationSet for the input partition. If the partition doesn't
 // exist or there are no healthy owners for the partition, an error is returned.
-func (r *PartitionInstanceRing) GetReplicationSetForPartitionAndOperation(partitionID int32, op Operation) (ReplicationSet, error) {
+// pickHighestPreferrablyNonReadOnlyOwnerFromEachZone instructs to pick the highest instance from each zone when building the replication set,
+// non-read-only instances are preferred over read-only instances (this is more important than picking the highest instance).
+func (r *PartitionInstanceRing) GetReplicationSetForPartitionAndOperation(partitionID int32, op Operation, pickHighestPreferrablyNonReadOnlyOwnerFromEachZone bool) (ReplicationSet, error) {
```

Comment:
[thought] When we'll merge this stuff to `main`, I think we should consider not breaking `GetReplicationSetForPartitionAndOperation()` to not clutter it with options that only make sense when `multiplePartitions = true`. I think an option may be creating a new data type `MultiPartitionInstanceRing` that embeds `PartitionInstanceRing` data structure, where we override the `GetReplicationSetForPartitionAndOperation()` function and it calls the "underlying" `getReplicationSetForPartitionAndOperation()` (lower case one).

Reply:
This is a function you added just for usage-tracker in the `usage-tracker` branch. But yes, maybe not cluttering and having a different structure would be great.